crates/uv-workspace/src/pyproject_mut.rs

```diff
@@ -503,13 +503,38 @@ pub fn add_dependency(
     deps: &mut Array,
     has_source: bool,
 ) -> Result<ArrayEdit, Error> {
-    // Find matching dependencies.
     let mut to_replace = find_dependencies(&req.name, Some(&req.marker), deps);
-    match to_replace.as_slice() {
+
+    let edit = match to_replace.as_slice() {
         [] => {
+            // determine the dependency list is sorted prior to
+            // adding the new dependency; the new dependency list
+            // will be sorted only when the original list is sorted
+            // so that users' custom dependency ordering is preserved.
+            let sorted = deps
+                .clone()
+                .into_iter()
+                .collect::<Vec<_>>()
+                .windows(2)
+                .all(|w| w[0].to_string() <= w[1].to_string());
```

Comment:
This should only pass when all values are strings. If they aren't the table is invalid; We could error, but for simplicity here it makes more sense to treat invalid files as unsorted, add our requirement to the end of the list and let it fail later when we try to use the pyproject.toml